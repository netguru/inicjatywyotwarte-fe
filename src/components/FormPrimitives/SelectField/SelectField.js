import React from 'react'
import styled from 'styled-components'
import Label from 'components/FormPrimitives/FieldLabel/FieldLabel'
import dropdown_arrow from 'assets/dropdown_arrow.png'

const InputBox = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 1.5rem;
  width: 100%;
`
InputBox.displayName = "InputBox";

const Select = styled.select`
  height: 3rem;
  font-size: 0.9rem;
  padding: 0 0.8rem;
  font-weight: 100;
  background: ${props => props.theme.lightBackgroundColor};
  border: 1px solid ${props => props.theme.modalColorVeryLight};
  border-radius: ${props => props.theme.borderRadiusSmall};
  outline: none;
  color: ${props => props.theme.modalColorLight};

  &:focus,
  &:active {
    border: ${props => props.theme.inputActiveBorder};
  }

  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: url(${dropdown_arrow}) no-repeat right;
  background-color: ${props => props.theme.lightBackgroundColor};
  background-size: 1.2rem;
  background-position-x: 95%;
  cursor: pointer;
`
Select.displayName = "Select";

export default function SelectField({
  name,
  options,
  label
}) {
  return (
    <InputBox>
      <Label htmlFor={name}>{label}</Label>
      <Select id={name}>
        {options.codes.map(option => {
          return (
            <option key={option} value={option}>
              {options.getText(option)}
            </option>
          )
        })}
      </Select>
    </InputBox>
  )
}
