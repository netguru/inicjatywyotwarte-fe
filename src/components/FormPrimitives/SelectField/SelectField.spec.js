import React from 'react'
import SelectField from './SelectField'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const baseProps = {
  name: 'name',
  options: {
    codes: [1, 2, 3],
    getText: jest.fn()
  },
  label: 'label'
}

describe('SelectField', () => {
  test('renders select field', () => {
    const wrapper = shallow(<SelectField {...baseProps} />);
    expect(wrapper.find('InputBox').length).toEqual(1);
    expect(wrapper.find('Select').children().length).toEqual(baseProps.options.codes.length)
  });

  test('match snapshot', () => {
    const wrapper = shallow(<SelectField {...baseProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});


