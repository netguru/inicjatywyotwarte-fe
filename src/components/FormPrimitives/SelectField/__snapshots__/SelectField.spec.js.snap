// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField match snapshot 1`] = `
<InputBox>
  <styled.label
    htmlFor="name"
  >
    label
  </styled.label>
  <Select
    id="name"
  >
    <option
      key="1"
      value={1}
    />
    <option
      key="2"
      value={2}
    />
    <option
      key="3"
      value={3}
    />
  </Select>
</InputBox>
`;
